osu.Game/Beatmaps/Formats/LegacyBeatmapEncoder.cs

```diff
@@ -58,7 +60,7 @@ private void handleGeneral(TextWriter writer)
         {
             writer.WriteLine("[General]");
 
-            writer.WriteLine(FormattableString.Invariant($"AudioFilename: {Path.GetFileName(beatmap.Metadata.AudioFile)}"));
+            if (beatmap.Metadata.AudioFile != null) writer.WriteLine(FormattableString.Invariant($"AudioFilename: {Path.GetFileName(beatmap.Metadata.AudioFile)}"));
```

Comment:
should this not be `virtual` instead?

Reply:
Can do, would need to also make it by default inside `BeatmapMetadata` too. Sound fine?

Reply:
hmm... not sure about that (do we want to use it going forward?)

Reply:
Seems like stable isn't setting virtual anymore, and it behaves fine without it.